### PR TITLE
Feature/menu drawer

### LINKED ISSUE #52: Base of App
the core navigation through the app -> navigation slide on the left (menu)

### PR DESCRIPTION
**Pull request by andreBrauni**
fix #52

### CODE REVIEW

File: lust/lib/pages/statefulPage.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import 'package:flutter/material.dart';
+
+abstract class StatefulPage extends StatefulWidget {
```
[the remark this answers is not in the document]
Comment (andreBrauni) [reply lust/lib/pages/statefulPage.dart]: Well, our Capacitypage has to inherit from  StatefulWidget in the end and i want it to also inherit the title and the icon from somewhere.
Is there a way to extend multiple super classes? or any other way except from extending to get that? (i dont understand the "mixin" and "factory" thing in dart yet.

File: lust/lib/pages/statefulPage.dart [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+  StatefulPage({this.title, this.icon});
+
+  final String title;
+  final icon;
+}
```
[the remark this answers is not in the document]
Comment (andreBrauni) [reply lust/lib/pages/statefulPage.dart]: done

File: lust/lib/pages/capacityPage.dart [hunk 3]
```diff
@@ -1,14 +1,29 @@
...
+
 import 'package:lust/widgets/capacityPage/capacityGraph.dart';
 import 'package:lust/widgets/capacityPage/capacityInfo.dart';
 
-class CapacityPage extends StatefulWidget {
+import 'package:lust/widgets/utils/menuDrawer.dart';
+
+class CapacityPage extends StatefulPage {
+
+  CapacityPage({this.title = "Capacity", this.icon = Icons.equalizer});
```
[the remark this answers is not in the document]
Comment (andreBrauni) [reply lust/lib/pages/capacityPage.dart]: done

File: lust/lib/widgets/utils/menuDrawer.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      title: Text(page.title),
+    );
+  }
+}
+
+// For mocking user info
+class MockUserInfo {
+  const MockUserInfo({this.name, this.email});
+
+  final String name;
```
[the remark this answers is not in the document]
Comment (andreBrauni) [reply lust/lib/widgets/utils/menuDrawer.dart]: done

File: lust/lib/widgets/utils/menuDrawer.dart [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      // menu item CapacityPage
+      onTap: () => _switchPage(context, page),
+      leading: Icon(page.icon),
+      title: Text(page.title),
+    );
+  }
+}
+
+// For mocking user info
+class MockUserInfo {
```
[the remark this answers is not in the document]
Comment (andreBrauni) [reply lust/lib/widgets/utils/menuDrawer.dart]: dont agree

File: lust/lib/pages/pageContainer.dart [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import 'package:flutter/material.dart';
+
+class PageContainer {
+  final String title;
+  final IconData icon;
+  final Widget pageObject;
+
+  PageContainer(this.title, this.icon, this.pageObject);
+}
```
Comment (worldpotato) [inline lust/lib/pages/pageContainer.dart]: add one empty line

File: lust/test/widget_tests/menuDrawer_test.dart [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  // Verify there is a tile for the page given by name
+  expect(
+      find.descendant(of: find.byType(ListView), matching: find.text(page.title)), findsOneWidget);
+  // Open the corresponding page
+  await tester.tap(find.descendant(of: find.byType(ListView), matching: find.text(page.title)));
+  await tester.pump();
+  // Verify the page was opened
+  expect(find.byType(page.runtimeType), findsOneWidget);
+}
+*/
```
Comment (worldpotato) [inline lust/test/widget_tests/menuDrawer_test.dart]: pls add some empty line

File: lust/lib/pages/pageContainer.dart (hunk 6 above)
Comment (andreBrauni) [reply lust/lib/pages/pageContainer.dart]: done

File: lust/test/widget_tests/menuDrawer_test.dart (hunk 7 above)
Comment (andreBrauni) [reply lust/test/widget_tests/menuDrawer_test.dart]: done

**Pull request merged**